app/src/page/file/view.js

```diff
@@ -96,8 +102,30 @@ class FilePage extends React.PureComponent {
     }
   }
 
+  componentWillReceiveProps(nextProps) {
+    const { purchasedUris: prevPurchasedUris, navigation, contentType } = this.props;
+    const { uri } = navigation.state.params;
+    const { fileInfo, purchasedUris, failedPurchaseUris, streamingUrl } = nextProps;
+
+    if (this.state.downloadPressed && failedPurchaseUris.includes(uri)) {
+      this.setState({ downloadPressed: false, fileViewLogged: false, mediaLoaded: false });
+    }
+
+    if (prevPurchasedUris.length != purchasedUris.length && NativeModules.UtilityModule) {
+      NativeModules.UtilityModule.checkDownloads();
+    }
+
+    const mediaType = Lbry.getMediaType(contentType);
+    const isPlayable = mediaType === 'video' || mediaType === 'audio';
+    if (fileInfo && fileInfo.streaming_url && !this.state.streamingMode && isPlayable) {
```

Comment:
this looks like it should be something like below (unless you intended 2 `setState` calls to be called in succession sometimes, but even in that case you can abstract the shared part of the if statement):

```
if (!this.state.streamingMode && isPlayable) {
  if (streamingUrl) {
    this.setState({ streamingMode: true, currentStreamUrl: streamingUrl });
  }
  else if (fileInfo && fileInfo.streaming_url) {
    this.setState({ streamingMode: true, currentStreamUrl: fileInfo.streaming_url });
  }
}
```